### PR TITLE
Add dyd file writer

### PR DESCRIPTION
**Pull request by lecourtoisflo**
Linked dynawo PR 1176
Unit test for DYD writer will be done in another PR

### CODE REVIEW

File: sources/Algo/include/Algo.h [hunk 1]
```diff
@@ -133,16 +134,27 @@ struct GeneratorDefinition {
...
   };
 
+  using ReactiveCurvePoint = DYN::GeneratorInterface::ReactiveCurvePoint;  ///< Alias for reactive curve point
+
   /**
    * @brief Constructor
    *
    * @param genId generator id
    * @param type the model to use
+   * @param nodeid the node id connected to the generator
```
Comment (marcochiaramello) [inline sources/Algo/include/Algo.h:144]: typo: nodeId

File: sources/Outputs/include/Dyd.h [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+   * Create maro connectors elements for generators and loads
+   *
+   * @returns list of macro connectors
+   */
+  static std::vector<boost::shared_ptr<dynamicdata::MacroConnector>> writeMacroConnectors();
+
+  /**
+   * @brief Write macro static references
+   *
+   * Create maro static reference elements for generators and loads
```
Comment (marcochiaramello) [inline sources/Outputs/include/Dyd.h:124]: typo : macro

**Review by marcochiaramello — CHANGES_REQUESTED** [review]

File: sources/Algo/include/Algo.h (hunk 1 above)
Comment (lecourtoisflo) [reply sources/Algo/include/Algo.h:144]: No, in the parameters, it is nodeid

File: sources/Outputs/include/Dyd.h (hunk 2 above)
Comment (lecourtoisflo) [reply sources/Outputs/include/Dyd.h:124]: OK

File: sources/Algo/include/Algo.h (hunk 1 above)
Comment (rosiereflo) [reply sources/Algo/include/Algo.h:144]: Please respect the coding style independently from the parameters

File: sources/Algo/include/Algo.h [hunk 3]
```diff
@@ -182,9 +194,12 @@ struct LoadDefinition {
    * @brief Constructor
    *
    * @param loadId the load id
+   * @param nodeid the node id connected to the load
    */
-  explicit LoadDefinition(const inputs::Load::LoadId& loadId) : id{loadId} {}
-  inputs::Load::LoadId id;  ///< load id
+  explicit LoadDefinition(const inputs::Load::LoadId& loadId, const inputs::Node::NodeId& nodeid) : id{loadId}, nodeId{nodeid} {}
```
Comment (rosiereflo) [inline sources/Algo/include/Algo.h:199]: same as above

File: sources/Outputs/src/Job.cpp [hunk 4]
```diff
@@ -51,7 +51,7 @@ const std::string Job::solverFilename_ = "solver.par";
 const std::string Job::solverName_ = "dynawo_SolverSIM";
 const std::string Job::solverParId_ = "SimplifiedSolver";
 
-Job::Job(const JobDefinition& def) : def_{def} {}
+Job::Job(JobDefinition&& def) : def_{std::forward<JobDefinition>(def)} {}
```
Comment (rosiereflo) [inline sources/Outputs/src/Job.cpp:54]: Why is this needed? or is it just for the beauty of using universal reference? ;)

File: sources/Outputs/include/Dyd.h [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+   * @brief Dyd definition to provide informations to build the Dyd file
+   */
+  struct DydDefinition {
+    /**
+     * @brief Constructor
+     *
+     * @param base the basename for current file (corresponds to filepath basename)
+     * @param filepath the filepath of the dyd file to write
+     * @param gens generators definition coming from algorithms
+     * @param loaddefs laod definitions coming from algorithms
```
Comment (rosiereflo) [inline sources/Outputs/include/Dyd.h:50]: typo load

File: sources/Outputs/include/Dyd.h [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+   *
+   * @param def the generator definition to process
+   *
+   * @returns the macro connection element
+   */
+  static std::vector<boost::shared_ptr<dynamicdata::MacroConnect>> writeGenConnect(const algo::GeneratorDefinition& def, unsigned int index);
+
+ private:
+  static const std::unordered_map<algo::GeneratorDefinition::ModelType, std::string>
+      correspondence_lib_;  ///< Correspondence between generator model type and libary name in dyd file
```
Comment (rosiereflo) [inline sources/Outputs/include/Dyd.h:154]: typo library

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: sources/Outputs/src/Job.cpp (hunk 4 above)
Comment (lecourtoisflo) [reply sources/Outputs/src/Job.cpp:54]: Not "needed", but because of the way we initialize the job in the context, it will be faster with universal references (move instead of copy)

File: sources/Outputs/include/Dyd.h (hunk 5 above)
Comment (lecourtoisflo) [reply sources/Outputs/include/Dyd.h:50]: OK

File: sources/Outputs/include/Dyd.h (hunk 6 above)
Comment (lecourtoisflo) [reply sources/Outputs/include/Dyd.h:154]: OK

File: sources/Algo/include/Algo.h (hunk 3 above)
Comment (lecourtoisflo) [reply sources/Algo/include/Algo.h:199]: OK

File: sources/Inputs/src/NetworkManager.cpp [hunk 7]
```diff
@@ -69,7 +69,10 @@ NetworkManager::buildTree() {
         // node should exist at this point
         assert(nodes_.count(nodeid));
 #endif
-        nodes_[nodeid]->generators.emplace_back((*it_g)->getID());
+        if ((*it_g)->isVoltageRegulationOn()) {
+          // We don't use dynamic models for generators without voltage regulation
```
Comment (marcochiaramello) [inline sources/Inputs/src/NetworkManager.cpp:73]: small fix on the comment: "We don't use dynamic models for generators with voltage regulation disabled"

File: sources/Algo/include/Algo.h (hunk 1 above)
Comment (lecourtoisflo) [reply sources/Algo/include/Algo.h]: OK

**Review by rosiereflo — APPROVED** [review]

**Review by cossartque — APPROVED** [review]

**Review by marcochiaramello — CHANGES_REQUESTED** [review]

File: sources/Inputs/src/NetworkManager.cpp (hunk 7 above)
Comment (lecourtoisflo) [reply sources/Inputs/src/NetworkManager.cpp:73]: OK

**Review by marcochiaramello — APPROVED** [review]

**Review by gautierbureau — APPROVED** [review]

File: sources/Outputs/include/Dyd.h [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+        filename{filepath},
+        generators{gens},
+        loads{loaddefs},
+        slackNode{slacknode} {}
+
+    std::string basename;                               ///< basename for file
+    std::string filename;                               ///< filepath for file to write
+    std::vector<algo::GeneratorDefinition> generators;  ///< generators found
+    std::vector<algo::LoadDefinition> loads;            ///< list of loads
+    std::shared_ptr<inputs::Node> slackNode;            ///< list of generators
```
Comment (PerninGuillaume) [inline sources/Outputs/include/Dyd.h:65]: Should it be in the comment the list of generators or the slack node to use ?

Comment (lecourtoisflo) [reply sources/Outputs/include/Dyd.h:65]: OK
